### ISSUE TITLE
Omitted fields are still populated in GeoODK forms

### ISSUE THREAD

**Issue by linzjax** [Cadasta/django-jsonattrs#2]
Currently, there's the option to have an omit column in a form which determines whether or not an attribute is defined in the schema. This creates some oddities because the field is not omitted from a GeoODK form, and if a user fills it out it throws an error.

<img width="1040" alt="screen shot 2016-06-29 at 11 32 05 am" src="https://cloud.githubusercontent.com/assets/8128188/16458473/2cf258fc-3ded-11e6-8c79-acf970ac1ee7.png">

```
location = SpatialUnitFactory.create(
            project=project,
            name='Captain America',
            attributes={
                'quality': 'text',
                'notes': 'does this work?'
            }
        )
        assert location.attributes['notes'] == 'Test notes'
```

```
self = {'quality': 'text'}, key = 'notes'

    def _check_key(self, key):
        self.setup_schema()
        if key not in self._attrs:
>           raise KeyError(key)
E           KeyError: 'notes'
```

I'm not sure I understand the purpose of the omit column. If a user didn't want a field included, wouldn't they just leave it out?

**Comment by ian-ross** [comment]
The `omit` field for attributes is used for schema composition, which is a feature that seemed like a good idea at the time but I don't know if we'll use.

Suppose that you have a `Party` model with a JSON attribute field.  You might have schemas for all parties (a platform-wide default), all parties within an organisation, all parties within a project within an organisation, and for a specific questionnaire for the project.  (These are set up by giving values for a prefix of the schema selectors chosen for the content type.)  The effective attribute schema for a given party is found by composing this series of schemas, starting from the platform-wide default (least specific), then going in order of increasing specificity, ending up with the schema with values for all of the schema selectors.

With this approach, it's easy to add new fields for more specific cases, and to override field types, choices, defaults, etc. for more specific cases, but what do you do if there's a field in the default schema that you don't want?  That's what the `omit` field is for.  It just flags that an attribute inherited from a less specific schema should be left out of the effective schema resulting from the schema composition.

Re-reading that, it sounds really complicated.  It's not really though, and it would have been _completely_ clear and transparent if I had got around to writing some documentation!

In summary though, you should probably just ignore the `omit` field for now since we're not doing any schema composition (as far as I know).

**Comment by ian-ross** [comment]
@linzjax Can I close this?  Does the description above make sense?  I will at some point get round to writing documentation (promise!).
